register.go

```diff
@@ -0,0 +1,91 @@
+package currency
+
+import (
+	"fmt"
+)
+
+// EmptyCurrencyCodeError indicates that the currency code was empty.
+type EmptyCurrencyCodeError struct{}
+
+func (e EmptyCurrencyCodeError) Error() string {
+	return "register currency error: empty currency code"
+}
+
+// CurrencyAlreadyExistsError indicates that the currency code already exists in the ISO list.
+type CurrencyAlreadyExistsError struct {
+	Code string
+}
+
+func (e CurrencyAlreadyExistsError) Error() string {
+	return fmt.Sprintf("register currency error: code %q already exists in ISO list", e.Code)
+}
+
+// RegisterCurrencyOptions defines parameters for registering a new or custom currency.
+type RegisterCurrencyOptions struct {
+	// NumericCode is usually a three-digit code, for example "999".
+	NumericCode string
```

Comment:
We also need a CountryCode here (first in the struct, above NumericCode), to provide data for `currency.ForCountryCode`